tests/lib/rules/no-unused-vars.js

```diff
@@ -333,6 +345,21 @@ ruleTester.run("no-unused-vars", rule, {
             ]
         },
 
+        // Using object rest for variable omission
+        {
+            code: "const data = { type: 'coords', x: 1, y: 2 };\nconst { type } = data;",
```

Comment:
I'm confused by this testcase -- the comment indicates that it's using object rest, but it doesn't appear to do that.